lib/src/material_color_picker.dart

```diff
@@ -77,47 +77,48 @@ class _MaterialColorPickerState extends State<MaterialColorPicker> {
     });
   }
 
-  ColorSwatch _findMainColor(Color shadeColor) {
-    for (final mainColor in _colors)
+  ColorSwatch? _findMainColor(Color shadeColor) {
+    for (final ColorSwatch? mainColor in _colors!)
       if (_isShadeOfMain(mainColor, shadeColor)) return mainColor;
 
-    return (shadeColor is ColorSwatch && _colors.contains(shadeColor))
+    return (shadeColor is ColorSwatch && _colors!.contains(shadeColor))
         ? shadeColor
         : null;
   }
 
-  bool _isShadeOfMain(ColorSwatch mainColor, Color shadeColor) {
-    for (final shade in _getMaterialColorShades(mainColor)) {
+  bool _isShadeOfMain(ColorSwatch? mainColor, Color? shadeColor) {
+    for (final shade in _getMaterialColorShades(mainColor!)) {
       if (shade == shadeColor) return true;
     }
     return false;
   }
 
-  void _onMainColorSelected(ColorSwatch color) {
+  void _onMainColorSelected(ColorSwatch? color) {
     var isShadeOfMain = _isShadeOfMain(color, _shadeColor);
-    final shadeColor = isShadeOfMain ? _shadeColor : (color[500] ?? color[400]);
+    final shadeColor =
+        isShadeOfMain ? _shadeColor : (color![500] ?? color[400]);
 
     setState(() {
       _mainColor = color;
       _shadeColor = shadeColor;
       _isMainSelection = false;
     });
-    if (widget.onMainColorChange != null) widget.onMainColorChange(color);
-    if (widget.onlyShadeSelection && !_isMainSelection) {
+    if (widget.onMainColorChange != null) widget.onMainColorChange?.call(color);
+    if (widget.onlyShadeSelection && !(_isMainSelection ?? false)) {
       return;
     }
     if (widget.allowShades && widget.onColorChange != null)
-      widget.onColorChange(shadeColor);
+      widget.onColorChange?.call(shadeColor!);
   }
 
   void _onShadeColorSelected(Color color) {
     setState(() => _shadeColor = color);
-    if (widget.onColorChange != null) widget.onColorChange(color);
+    if (widget.onColorChange != null) widget.onColorChange?.call(color);
```

Comment:
Same here -- you don't need a null check followed by `?.call`. You can remove the null check.